tests/integration_tests/cors_headers_test.rs

```diff
@@ -0,0 +1,228 @@
+use std::{collections::HashMap, time::Duration};
+
+use crate::helpers::{ExpectedResponse, TestEnv, check_response, retry_async};
+use anyhow::{Context, anyhow};
+use candid::Principal;
+use http::{Method, StatusCode};
+use reqwest::{Client, Request};
+use tokio::runtime::Runtime;
+use tracing::info;
+use url::Url;
+
+const REQUEST_RETRY_TIMEOUT: Duration = Duration::from_secs(20);
+const REQUEST_RETRY_INTERVAL: Duration = Duration::from_secs(2);
+
+// Test scenario:
+// - install counter canister
+// - make various HTTP requests OPTIONS/GET/POST and verify the CORS headers of the responses
+
+pub fn cors_headers_test(env: &TestEnv) -> anyhow::Result<()> {
+    // some canister ID and subnet ID for testing -- they need to be valid principals, but don't need to exist
+    // as OPTIONS requests are directly replied to by the IC gateway and all other requests will just return a 400
+    // but the IC gateway still sets the CORS headers.
+    let canister_id =
+        Principal::from_text("rwlgt-iiaaa-aaaaa-aaaaa-cai").expect("failed to parse canister id");
```

Comment:
just fyi i wrote a tiny macro that makes it easier: https://github.com/dfinity/ic-gateway/blob/9b5702b16c651af631c32e54fc401ab6fd04f010/src/routing/domain.rs#L17

```
use ic_gateway::principal;

let canister_id = principal!("rwlgt-iiaaa-aaaaa-aaaaa-cai");
```